packages/extension/src/contexts/NetworkContext/NetworkContext.tsx

```diff
@@ -55,19 +55,25 @@ const NetworkProvider: FC = ({ children }) => {
   useEffect(() => {
     let retryCount = 0;
     const maxRetries = 3;
+    let nodeIndex = 0;
 
     const connectToNetwork = async () => {
       const network = loadNetwork();
       setNetworkName(network.name);
+      let node = network.server;
+      if (nodeIndex !== 0 && network.nodes?.[nodeIndex]) {
+        node = network.nodes[nodeIndex];
+      }
       try {
-        const ws = await connectToLedger(network.server);
+        const ws = await connectToLedger(node);
         setClient(ws);
       } catch (err) {
         setClient(null);
         setIsConnectionFailed(true);
         // If connect fails, retry up to maxRetries times
         if (retryCount < maxRetries) {
           retryCount += 1;
+          nodeIndex += 1;
           setTimeout(connectToNetwork, 1000);
         }
```

Comment:
if (retryCount < maxRetries) {
 retryCount += 1;
} else {
 nodeIndex += 1;
}

That way we retry the same server at the index twice because switching to another one

Reply:
The algorithm is gonna change
It's not ready for review